tracer/src/Datadog.Trace.BenchmarkDotNet/DatadogExporter.cs

```diff
@@ -5,185 +5,242 @@
+    public static readonly DatadogExporter Default = new();
 
-        static DatadogExporter()
+    static DatadogExporter()
+    {
+        try
         {
-            try
-            {
-                Environment.SetEnvironmentVariable(Configuration.ConfigurationKeys.CIVisibility.Enabled, "1", EnvironmentVariableTarget.Process);
-            }
-            catch
-            {
-                // .
-            }
+            Environment.SetEnvironmentVariable(Configuration.ConfigurationKeys.CIVisibility.Enabled, "1", EnvironmentVariableTarget.Process);
         }
-
-        /// <inheritdoc />
-        public string Name => nameof(DatadogExporter);
-
-        /// <inheritdoc />
-        public void ExportToLog(Summary summary, ILogger logger)
+        catch
         {
+            // .
         }
+    }
 
-        /// <inheritdoc />
-        public IEnumerable<string> ExportToFiles(Summary summary, ILogger consoleLogger)
-        {
-            CIVisibility.Initialize();
-            DateTimeOffset startTime = DateTimeOffset.UtcNow;
-            Exception exception = null;
+    private DatadogExporter()
+    {
+        TestSession = TestSession.GetOrCreate(Environment.CommandLine, Environment.CurrentDirectory, "BenchmarkDotNet", DateTime.UtcNow, false);
+        TestModule = TestSession.CreateModule(Assembly.GetEntryAssembly().GetName().Name, "BenchmarkDotNet", typeof(IDiagnoser).Assembly.GetName().Version.ToString());
+        LifetimeManager.Instance.AddAsyncShutdownTask(
+            async () =>
+            {
+                await TestModule.CloseAsync().ConfigureAwait(false);
+                TestSession.Tags.CommandExitCode = Environment.ExitCode.ToString();
+                await TestSession.CloseAsync(Environment.ExitCode == 0 ? TestStatus.Pass : TestStatus.Fail).ConfigureAwait(false);
+            });
+    }
+
+    /// <inheritdoc />
+    public string Name => nameof(DatadogExporter);
+
+    internal TestSession TestSession { get; }
 
-            try
+    internal TestModule TestModule { get; }
+
+    /// <inheritdoc />
+    public void ExportToLog(Summary summary, ILogger logger)
+    {
+    }
+
+    /// <inheritdoc />
+    public IEnumerable<string> ExportToFiles(Summary summary, ILogger consoleLogger)
+    {
+        var testSession = TestSession;
+        Dictionary<Type, TestSuiteWithEndDate> testSuites = new();
+        Exception exception = null;
+
+        try
+        {
+            foreach (var report in summary.Reports)
             {
-                Tracer tracer = Tracer.Instance;
+                var benchmarkStartDate = DateTime.MinValue;
+                var benchmarkEndDate = DateTime.MinValue;
 
-                foreach (var report in summary.Reports)
+                if (report?.BenchmarkCase?.Descriptor is { Type: { } type } descriptor && summary.HostEnvironmentInfo is { CpuInfo.Value: { } cpuInfo } hostEnvironmentInfo)
```

Comment:
Can we either enable nullable in this file or for the whole project (ideally)

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3774/commits/5a583c087ff9f055bfb81e21ede8862247db8c3b